### PR TITLE
[LT] Merge cache.h

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67929
* #67766

Summary:
1. Write a node-hash based unit test for Cache
2. Replace CHECK with TORCH_CHECK in IrUtil

Differential Revision: [D32246134](https://our.internmc.facebook.com/intern/diff/D32246134)

### CODE REVIEW

File: torch/csrc/lazy/core/ir_util.cpp [hunk 1]
```diff
@@ -25,22 +25,26 @@ std::vector<torch::lazy::Node*> Util::ComputePostOrder(
...
+            "Graph loop found at ",
+            output.node->ToString());
       }
       (*emap)[node] = kEmitted;
       // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
       post_order.push_back(const_cast<torch::lazy::Node*>(node));
       queue.pop_back();
     } else {
-      DCHECK_EQ(it->second, kEmitted);
+      CHECK_EQ(it->second, kEmitted);
```
Comment (alanwaketan) [inline torch/csrc/lazy/core/ir_util.cpp:47]: No `TORCH_CHECK`?

File: torch/csrc/lazy/core/cache.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      DoLRU(emplace_result.first->second);
+    } else if (element_list_.size() > max_size_) {
+      Element* last = &element_list_.back();
+      element_map_.erase(&last->first);
+      element_list_.pop_back();
+    }
+    return emplace_result.first->second->second;
+  }
+
+  // Retrieves the existing object if it exists. If it does, it's position in
```
Comment (wconstab) [inline torch/csrc/lazy/core/cache.h:47]: it's -> its

**Review by wconstab — APPROVED** [review]

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32246134).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32246134).

**Comment by desertfire** [comment]
@desertfire has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32246134).

**Pull request closed without merging**